multi_import/multi_importer.py

```diff
@@ -85,9 +85,20 @@ def import_data(self, data):
 
         bound_importers = self._transform_multi_input(data)
 
-        for importer, filename, dataset in bound_importers:
-            result = importer.import_data(dataset, context, transaction=False)
-            results.add_result(filename, result)
+        for importer, datasets in bound_importers:
+            serializer_context = importer.get_serializer_context(context)
+
+            read_datasets = [
+                (filename, importer.read_rows(dataset))
+                for filename, dataset in datasets
+            ]
+
+            for filename, rows in read_datasets:
+                importer.load_instances(rows, serializer_context)
+
+            for filename, rows in read_datasets:
+                result = importer.process_rows(rows, serializer_context)
+                results.add_result(filename, result)
```

Comment:
**Scenario**

The user imports 2 files, for the same model, each with a row that have identical universal IDs. There is no model instance in the database that matches these rows.

**Current behaviour**

We process file 1:

* `load_instance` fails to find a matching instance
* `process_row` creates a new instance

Then we process file 2:

* `load_instance` matches the instance created in the first file
* `process_row` updates that instance

**Desired behaviour**

Both rows should attempt to create a new instance, because there was no matching instance prior to the import. This will result in a validation error stating that you can't create duplicate instances.

This validation error occurs if the 2 rows are in the same file.

This change ensures multiple files of the same model, are processed as though they are 1 file.